lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -579,6 +579,27 @@ void CTimeSeriesDecompositionDetail::CPeriodicityTest::handle(const SNewComponen
     // components only when we test.
 }
 
+bool CTimeSeriesDecompositionDetail::CPeriodicityTest::shouldTest(ETest test,
+                                                                  core_t::TTime time) const {
+    // We need to test more frequently than we compress because it
+    // only happens each 336 buckets and would significantly delay
```

Comment:
Where is this `336` coming from? It would be nice to explain that too in the comment.

Reply:
Added an explanation in [this](https://github.com/elastic/ml-cpp/pull/218/commits/bd8854ed218330671b07160db35c161c76a312d4) commit.